docs/content/querying/lookups.md

```diff
@@ -323,6 +323,8 @@ It is possible to save the configuration across restarts such that a node will n
 |Property|Description|Default|
 |--------|-----------|-------|
 |`druid.lookup.snapshotWorkingDir`| Working path used to store snapshot of current lookup configuration, leaving this property null will disable snapshot/bootstrap utility|null|
+|`druid.lookup.numLookupLoadingThreads`| Number of threads for loading the lookups in parallel on startup.This threadpool is created only at the time of start and destroyed once startup is done. It is not kept during the lifetime of the JVM.|10|
+|`druid.lookup.enableLookupSyncOnStartup`|Enable the lookup synchronization process with coordinator on startup.|false|
```

Comment:
- It requires much more elaborate description, for people who read about lookups for the first time
 - Doesn't Coordinator do the opposite? In `LookupCoordinatorManager`. How does this interoperate? @himanshug @drcrallen do you have thoughts on this design?

Reply:
i worked with @a2l007 for the design of this feature. its disabled by default to have no behavioral change e.g. peons/historicals suddenly start taking alot of time, so this is enabled by user consciously. also most users who don't use lookups as of now and default enabling would make their peons/historicals to make one coordinator request at startup for no reason.
this property is for the nodes that consumer lookups (like peons/historicals/brokers) and irrelevant for coordinator which knows whether any lookups are configured or not (from DB) and then always propagates them if there are any.

Reply:
I mean, if this option is on, queryable node and coordinator will try to do the same thing, from opposite directions, if I understand correctly. Coordinator will try to push lookups to the queryable node, and also this node will try to pull the lookups from the coordinator itself. I wonder if those two processes could clash, or e. g. lead to duplicating lookups on the queryable node? And from which side it is more "correct" to do this - push from coordinator or pull from the queryable? Should there be an option to disable push from coordinator?

Reply:
no, coordinator can't push anything till LookupReferencesManager.start() finishes , those requests from coordinator would just fail and nothing bad would happen.
However coordinator wouldn't be able to see new lookup-node till its startup completes because lookup-node would get announced at the end of its lifecycle when LookupReferencesManager.start() is already done. so, coordinator wouldn't push anything while the syncing introduced here happens on the lookup-node.